libtsuba/include/tsuba/Range.h

```diff
@@ -0,0 +1,50 @@
+#ifndef KATANA_LIBTSUBA_TSUBA_RANGE_H_
+#define KATANA_LIBTSUBA_TSUBA_RANGE_H_
+
+#include <iterator>
+#include <type_traits>
+
+#include "katana/config.h"
+
+namespace tsuba {
+
+/**
+ * A StandardRange is a range over standard C++ begin and end iterators.
+ *
+ * Implements std::ranges::sized_range (C++20).
+ */
+template <typename Iterator>
+class StandardRange {
```

Comment:
Please add a TODO to move this class to `libsupport` and `katana` namespace eventually.